gravityflow/gravityflow
# Implement assignee factory

This PR refactors the way assignees are handled to allow new types of assignees to be implemented more easily. Its also adds two new filters: gravityflow_assignee_choices and gravityflow_inbox_search_criteria.

**Testing instructions**
1. Check that all steps work as expected for all assignee types.
2. Review to make sure it's backwards compatible with custom code and extensions.
3. Add the code in [this gist](https://gist.github.com/stevehenty/dc73ca8c9501d11f28419ce92399aa3b) to create the bare-bones, proof of concept for of a new Department Assignee. It will not add a field in the field editor - it will just make an option available in the Fields section of the assignee list in the step settings. This refactoring will allow us to add new fields to the form editor from custom code and extensions and handle the assignee logic differently for each. 
4. Add the Department field as an assignee to an approval step and a user input step.
5. Submit an entry to trigger the step. Find the entry in the inbox pending for the Department: Marketing assignee. Complete the steps.
